official/utils/misc/distribution_utils.py

```diff
@@ -35,28 +36,35 @@ def get_distribution_strategy(num_gpus,
     turn_off_distribution_strategy: when set to True, do not use any
       distribution strategy. Note that when it is True, and num_gpus is
       larger than 1, it will raise a ValueError.
+    distribute_strategy: If set and `turn_off_distribution_strategy` is False,
+      use the specified Distribution Strategy.
 
   Returns:
     tf.contrib.distribute.DistibutionStrategy object.
   Raises:
     ValueError: if turn_off_distribution_strategy is True and num_gpus is
     larger than 1
   """
-  if num_gpus == 0:
-    if turn_off_distribution_strategy:
-      return None
-    else:
+  if turn_off_distribution_strategy:
+    if num_gpus > 1:
+      raise ValueError("When {} GPUs are specified, "
+                       "turn_off_distribution_strategy flag cannot be set to"
+                       "True.".format(num_gpus))
+     return None
+
+  if distribute_strategy:
+    distribute_strategy = distribute_strategy.lower()
+  
+  if distribute_strategy == "one_device" or num_gpus <= 1:
+    if num_gpus == 0:
       return tf.contrib.distribute.OneDeviceStrategy("device:CPU:0")
-  elif num_gpus == 1:
-    if turn_off_distribution_strategy:
-      return None
     else:
+      if num_gpus > 1:
+        raise ValueError("`OneDeviceStrategy` can not be used for more than "
+                        "one device.")
       return tf.contrib.distribute.OneDeviceStrategy("device:GPU:0")
-  elif turn_off_distribution_strategy:
-    raise ValueError("When {} GPUs are specified, "
-                     "turn_off_distribution_strategy flag cannot be set to"
-                     "True.".format(num_gpus))
-  else:  # num_gpus > 1 and not turn_off_distribution_strategy
+
+  if distribute_strategy == "mirrored":
```

Comment:
hm, from what i can tell, this only ever returns a mirrored strategy when distribute_strategy = "mirrored". or did i miss something?

Reply:
Is this what we want? `distribute_strategy` override the strategy but if it is None, auto chooses one. See line 58.

Reply:
sorry i still don't see how `distribute_strategy=None` will ever return MirroredStrategy in this code. Line 58 is only about one device strategy it seems?

Reply:
Sorry I misread you comment. Added commit to return MS by default.